api/app/controllers/spree/api/v2/storefront/account/credit_cards_controller.rb

```diff
@@ -14,6 +14,11 @@ def show
               render_serialized_payload { serialize_resource(resource) }
             end
 
+            def destroy
+              card = resource.find(params[:credit_card_id])
```

Comment:
@aplegatt I believe `resource` will return the credit card object, you should not need to call find here again

Reply:
@damianlegawiec Somehow, `resource` is giving me an array of `Spree::CreditCard`s here. I'm going to check the finder maybe something is wrong there.

Reply:
@aplegatt yes, the `resource` method is wrong it should be

```ruby
def scope
  resource_finder.new.execute(scope: scope, params: params)
end

def resource
  scope.find(params[:id])
end
```

Reply:
@aplegatt 

also the `index` method is wront as well...

please replicate https://github.com/spree/spree/blob/a4efe63593e13d5e3147ba5995e72e008ff609a2/api/app/controllers/spree/api/v2/storefront/account/orders_controller.rb#L6

Reply:
It seems I'm getting a `SystemStackError: stack level too deep` with the `scope` method looking like that due to the recursion. But I was able to make finder work by adding this line: https://github.com/spree/spree/pull/10534/files#diff-69f5940e7246fe4920140f6901ac6242634718f5451586188111b5225ef26d36R7